wavefront_pyformance/runtime_metrics.py

```diff
@@ -0,0 +1,168 @@
+"""Python Runtime Metrics Collection Class."""
+
+import gc
+import multiprocessing
+import os
+import resource
+import threading
+
+import psutil
+
+
+class RuntimeCollector(object):
+    """Python Runtime Metrics Collection Class."""
+
+    # pylint: disable=E0012,R0205
+
+    def __init__(self, registry=None):
+        """Construct Runtime Metrics Collector."""
+        self.registry = registry
+        self.pid = os.getpid()
+        self.process = psutil.Process(self.pid)
+        self.pname = self.process.name()
+        self.status = self.process.status()
+        self.custom_tags = {"process_id": str(self.pid),
+                            "process_name": self.pname,
+                            "process_status": self.status}
+
+    def collect_cputimes(self):
+        """Collect CPU Times."""
+        cputimes = self.process.cpu_times()
+        system_cpu = cputimes[1]  # system cpu
+        metric1 = self.registry.gauge("cpu.times",
+                                      tags=self.custom_tags)
+        metric1.set_value(system_cpu)
+
+    def collect_cpupercent(self):
+        """Collect CPU in Percentage."""
+        cpupercent = self.process.cpu_percent(interval=1)
+        metric1 = self.registry.gauge("cpu.percent",
+                                      tags=self.custom_tags)
+        metric1.set_value(cpupercent)
+
+    def collect_memoryusage(self):
+        """Collect Memory Usage."""
+        usage = self.process.memory_info()[0] / float(2 ** 20)
+        metric1 = self.registry.gauge("memory.rss.usage",
+                                      tags=self.custom_tags)
+        metric1.set_value(usage)
+
+    def collect_memorypercent(self):
+        """Collect Memory in Percentage."""
+        memorypercent = self.process.memory_percent(memtype="rss")
+        metric1 = self.registry.gauge("memory.rss.percent",
+                                      tags=self.custom_tags)
+        metric1.set_value(memorypercent)
+
+    def collect_threads(self):
+        """Collect Threading Metrics."""
+        counter = 0
```

Comment:
You may use tuple assignment, as you use further in your code
```
counter, alive, daemon = 0, 0, 0
```

Reply:
Done